dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/RelativePeriods.java

```diff
@@ -712,6 +719,15 @@ public PeriodType getHighestFrequencyPeriodType( List<Period> periods )
                 dynamicNames, format ) );
         }
 
+        if ( isLast10FinancialYears() )
+        {
+            periods.addAll( getRollingRelativePeriodList( financialPeriodType, LAST_10_FINANCIAL_YEARS,
```

Comment:
We are adding to the list twice here. Just wondering if this is expected...

Reply:
yes, it is. I am not allowed to create a new PeriodType like I did before. 
This is the critical method I have to use in FinancialPeriodType
@Override
    public List<Period> generateRollingPeriods( Date date )
    {
        return generateLast5Years( date );
    }

To extend the method signature means extend the interface, which is breaking change to lead me to lot of changes like I describe before (Last_10_YEARS story). That's why I have to call it twice and manipulate the date param by second call.
Sorry, I didn't found any other solution